app/routes/content.js

```diff
@@ -25,8 +25,43 @@ export default Ember.Route.extend(Analytics, ResetScrollMixin, SetupSubmitContro
         this.render(this.get('editMode') ? 'submit' : 'content');
     },
     model(params) {
-        if (params.edit) this.set('editMode', true);
-        return this.store.findRecord('preprint', params.preprint_id);
+        if (params.edit)
+            this.set('editMode', true);
+
+        const findPreprint = this
+            .store
+            .findRecord('preprint', params.preprint_id);
+
+        // If we're already on a branded page, stay there.
+        if (this.get('theme.isProvider'))
+            return findPreprint;
+
+        // If we're on the OSF
+        return findPreprint
```

Comment:
This redirect process from a `/:preprint_id` to a ` preprints/:provider/:preprint_id/ briefly flashes a `Page Not Found`.